redash/query_runner/snowflake.py

```diff
@@ -104,16 +104,47 @@ def run_query(self, query, user):
 
         return json_data, error
 
+    def __run_query_without_warehouse(self, query, user):
+        region = self.configuration.get('region')
+
+        # for us-west we don't need to pass a region (and if we do, it fails to connect)
+        if region == 'us-west':
+            region = None
+
+        connection = snowflake.connector.connect(
+            user=self.configuration['user'],
+            password=self.configuration['password'],
+            account=self.configuration['account'],
+            region=region
+        )
+
+        cursor = connection.cursor()
```

Comment:
At this is shared between both run query methods, let's extract it into its own method (`get_cursor`?).